vm/loader/page_table/src/x64.rs

```diff
@@ -30,14 +31,57 @@ pub const X64_LARGE_PAGE_SIZE: u64 = 0x200000;
 /// Number of bytes in a 1GB page for X64.
 pub const X64_1GB_PAGE_SIZE: u64 = 0x40000000;
 
+/// Maximum number of page tables created for an x64 identity map
+pub const PAGE_TABLE_MAX_COUNT: usize = 20;
+
+static_assertions::const_assert_eq!(
+    PAGE_TABLE_ENTRY_SIZE * PAGE_TABLE_ENTRY_COUNT,
+    X64_PAGE_SIZE as usize
+);
+const PAGE_TABLE_SIZE: usize = PAGE_TABLE_ENTRY_COUNT * PAGE_TABLE_ENTRY_SIZE;
+
+/// Maximum number of bytes needed to store an x64 identity map
+pub const PAGE_TABLE_MAX_BYTES: usize = PAGE_TABLE_MAX_COUNT * X64_PAGE_SIZE as usize;
+
 #[derive(Copy, Clone, PartialEq, Eq, IntoBytes, Immutable, KnownLayout, FromBytes)]
 #[repr(transparent)]
 pub struct PageTableEntry {
     pub(crate) entry: u64,
 }
 
-impl std::fmt::Debug for PageTableEntry {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+#[derive(Copy, Clone, Debug)]
+pub struct MappedRange {
```

Comment:
please add docs for pub things (we should probably enable remove the expect(missing_docs) in the top level lib so we get the right lints firing)